pkg/drivers/metal/metal.go

```diff
@@ -0,0 +1,481 @@
+// SPDX-License-Identifier: BSD-3-Clause
```

Comment:
This PR includes both `pkg/drivers/equinix-metal/equinix-metal.go` and this copy of the same file. 

Based on your PR description, let's stick with `metal` rather than `equinix-metal`.
This also affects the test file and the directory containing these files.